server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheContainerCountersConfigurationResource.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.clustering.infinispan.subsystem;
+
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.ReloadRequiredRemoveStepHandler;
+import org.jboss.as.controller.SimpleResourceDefinition;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.as.controller.services.path.ResolvePathHandler;
+
+/**
+ * CacheContainerCountersResource
+ *
+ * @author Vladimir Blagojevic 
+ * @since 9.2
+ */
+public class CacheContainerCountersConfigurationResource extends SimpleResourceDefinition {
+    static final PathElement PATH = PathElement.pathElement(ModelKeys.COUNTERS, ModelKeys.COUNTERS_NAME);
```

Comment:
You are using the same path for the configuration and for the concrete counters below. They should be separate.

Reply:
Indeed, but I am adding them to different parent path. Configurations are at:
```
/profile=clustered/subsystem=datagrid-infinispan/cache-container=clustered/configurations=CONFIGURATIONS/counters=COUNTERS
```
while counter instances are at:
```
/profile=clustered/subsystem=datagrid-infinispan/cache-container=clustered/counters=COUNTERS
```
A bit confusing but a result of our discusssion on infinispan-dev. See details at http://lists.jboss.org/pipermail/infinispan-dev/2017-November/017917.html